lib/sync.js

```diff
@@ -209,6 +226,12 @@ SyncApi.prototype.peek = function(roomId) {
         // roll backwards to diverge old state:
         peekRoom.addEventsToTimeline(messages.reverse(), true);
 
+        // set the pagination token. Make sure this happens after adding
+        // events to the timeline, otherwise it will get reset.
+        if (response.messages.start) {
+            peekRoom.oldState.paginationToken = response.messages.start;
```

Comment:
We can't do this here; it HAS to be before we add the events. See https://github.com/matrix-org/matrix-js-sdk/commit/bd4d8433abe8cae32f1b5b4ab6e41970de307f9b#diff-d1172eb3f936fc93dbd0b89cdf357205R459